spec/api/sul_bib/publications_api_spec.rb

```diff
@@ -269,49 +269,87 @@ def validate_authorship(pub_hash, submission)
+        post '/publications', with_isbn_hash.to_json, headers
+        id = Publication.last.id
+        put "/publications/#{id}", with_isbn_changed_doi.to_json, headers
+        expect(result['identifier'].size).to eq(3)
+        expect(result['identifier']).to include(
+          a_hash_including('type' => 'isbn', 'id' => '1177188188181'),
+          a_hash_including('type' => 'doi', 'url' => '18819910019-updated'),
+          a_hash_including('type' => 'SULPubId', 'url' => "#{Settings.SULPUB_ID.PUB_URI}/#{id}", 'id' => id.to_s)
+        )
+      end
+
+      it 'deletes an identifier from the db if it is not in the incoming json' do
+        post '/publications', with_isbn_hash.to_json, headers
+        id = Publication.last.id
+        put "/publications/#{id}", with_isbn_deleted_doi.to_json, headers
+        expect(result['identifier'].size).to eq(2)
+        expect(result['identifier']).to include(
+          a_hash_including('type' => 'isbn', 'id' => '1177188188181'),
+          a_hash_including('type' => 'SULPubId', 'url' => "#{Settings.SULPUB_ID.PUB_URI}/#{id}", 'id' => id.to_s)
+        )
+      end
     end
 
-    it 'deletes an identifier from the db if it is not in the incoming json' do
-      post '/publications', with_isbn_hash.to_json, headers
-      id = Publication.last.id
-      put "/publications/#{id}", with_isbn_deleted_doi.to_json, headers
-      expect(result['identifier'].size).to eq(2)
-      expect(result['identifier']).to include(
-        a_hash_including('type' => 'isbn', 'id' => '1177188188181'),
-        a_hash_including('type' => 'SULPubId', 'url' => "#{Settings.SULPUB_ID.PUB_URI}/#{id}", 'id' => id.to_s)
-      )
+    context 'fails' do
+      context 'pub not found' do
+        it 'returns 404' do
+          put '/publications/88888888888', with_isbn_hash.to_json, headers
+          expect(response.status).to eq(404)
+        end
+      end
+
+      context 'when existing pub already has' do
+        let(:id) { '1' }
+        before { allow(Publication).to receive(:find).with(id).and_return(publication) }
+
+        it 'been deleted' do
+          allow(publication).to receive(:deleted?).and_return(true)
+          put "/publications/#{id}", with_isbn_hash.to_json, headers
+          expect(response.status).to eq(410)
+        end
+        it 'sciencewire_id' do
+          allow(publication).to receive(:sciencewire_id).and_return(2)
+          put "/publications/#{id}", with_isbn_hash.to_json, headers
+          expect(response.status).to eq(403)
+        end
+        it 'pmid' do
+          allow(publication).to receive(:pmid).and_return(3)
+          put "/publications/#{id}", with_isbn_hash.to_json, headers
+          expect(response.status).to eq(403)
+        end
+        it 'wos_uid' do
+          allow(publication).to receive(:wos_uid).and_return(4)
+          put "/publications/#{id}", with_isbn_hash.to_json, headers
+          expect(response.status).to eq(403)
+        end
+      end
     end
   end
```

Comment:
:+1: to adding these specs; agree with those HTTP response codes.  It looks like this rounds out my earlier comments above about whether the non-cap provenance data is protected from PUTs.

Reply:
So, as a whole, it looks like I approve of this commit - 7d3708cbc91132c4895dba95302b75b3542862db